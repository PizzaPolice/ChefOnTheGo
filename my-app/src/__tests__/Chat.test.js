import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import Chat from '../components/chat.jsx'


it('ActiveChef component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

// This test is shallow rendering our Chat component
describe('ActiveChef Component', () => {
    it('ActiveChef should render without throwing an error', () => {
        expect(shallow(<Chat />).exists(<form className='Chat'></form>)).toBe(true)
      })
    });