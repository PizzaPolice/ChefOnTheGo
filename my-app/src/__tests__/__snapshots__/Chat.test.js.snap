// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Chat
    chat=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="style"
>
        <section
                className="chatmessages"
        >
                <section
                        className="chatboxright"
                >
                        <p
                                className="name"
                        >
                                chef_name
                        </p>
                        <p
                                className="message"
                        >
                                I am your chef today. What can i do for you?
                        </p>
                </section>
        </section>
        <div
                className="chatinput"
                name="chatinput"
        >
                <h4>
                        Enter your message here
                </h4>
                <textarea
                        className="chatbox"
                        cols="100"
                        rows="10"
                />
                <div
                        className="btn-group"
                >
                        <button
                                onClick={[Function]}
                        >
                                Chat! 
                        </button>
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="chatmessages"
>
            <section
                        className="chatboxright"
            >
                        <p
                                    className="name"
                        >
                                    chef_name
                        </p>
                        <p
                                    className="message"
                        >
                                    I am your chef today. What can i do for you?
                        </p>
            </section>
</section>,
          <div
            className="chatinput"
            name="chatinput"
>
            <h4>
                        Enter your message here
            </h4>
            <textarea
                        className="chatbox"
                        cols="100"
                        rows="10"
            />
            <div
                        className="btn-group"
            >
                        <button
                                    onClick={[Function]}
                        >
                                    Chat! 
                        </button>
            </div>
</div>,
        ],
        "className": "style",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="chatboxright"
>
              <p
                            className="name"
              >
                            chef_name
              </p>
              <p
                            className="message"
              >
                            I am your chef today. What can i do for you?
              </p>
</section>,
            "className": "chatmessages",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="name"
>
                  chef_name
</p>,
                <p
                  className="message"
>
                  I am your chef today. What can i do for you?
</p>,
              ],
              "className": "chatboxright",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "chef_name",
                  "className": "name",
                },
                "ref": null,
                "rendered": "chef_name",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I am your chef today. What can i do for you?",
                  "className": "message",
                },
                "ref": null,
                "rendered": "I am your chef today. What can i do for you?",
                "type": "p",
              },
            ],
            "type": "section",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Enter your message here
</h4>,
              <textarea
                className="chatbox"
                cols="100"
                rows="10"
/>,
              <div
                className="btn-group"
>
                <button
                                onClick={[Function]}
                >
                                Chat! 
                </button>
</div>,
            ],
            "className": "chatinput",
            "name": "chatinput",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enter your message here",
              },
              "ref": null,
              "rendered": "Enter your message here",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "chatbox",
                "cols": "100",
                "rows": "10",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  onClick={[Function]}
>
                  Chat! 
</button>,
                "className": "btn-group",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Chat! ",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Chat! ",
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="style"
>
          <section
                    className="chatmessages"
          >
                    <section
                              className="chatboxright"
                    >
                              <p
                                        className="name"
                              >
                                        chef_name
                              </p>
                              <p
                                        className="message"
                              >
                                        I am your chef today. What can i do for you?
                              </p>
                    </section>
          </section>
          <div
                    className="chatinput"
                    name="chatinput"
          >
                    <h4>
                              Enter your message here
                    </h4>
                    <textarea
                              className="chatbox"
                              cols="100"
                              rows="10"
                    />
                    <div
                              className="btn-group"
                    >
                              <button
                                        onClick={[Function]}
                              >
                                        Chat! 
                              </button>
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="chatmessages"
>
              <section
                            className="chatboxright"
              >
                            <p
                                          className="name"
                            >
                                          chef_name
                            </p>
                            <p
                                          className="message"
                            >
                                          I am your chef today. What can i do for you?
                            </p>
              </section>
</section>,
            <div
              className="chatinput"
              name="chatinput"
>
              <h4>
                            Enter your message here
              </h4>
              <textarea
                            className="chatbox"
                            cols="100"
                            rows="10"
              />
              <div
                            className="btn-group"
              >
                            <button
                                          onClick={[Function]}
                            >
                                          Chat! 
                            </button>
              </div>
</div>,
          ],
          "className": "style",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <section
                className="chatboxright"
>
                <p
                                className="name"
                >
                                chef_name
                </p>
                <p
                                className="message"
                >
                                I am your chef today. What can i do for you?
                </p>
</section>,
              "className": "chatmessages",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="name"
>
                    chef_name
</p>,
                  <p
                    className="message"
>
                    I am your chef today. What can i do for you?
</p>,
                ],
                "className": "chatboxright",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "chef_name",
                    "className": "name",
                  },
                  "ref": null,
                  "rendered": "chef_name",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "I am your chef today. What can i do for you?",
                    "className": "message",
                  },
                  "ref": null,
                  "rendered": "I am your chef today. What can i do for you?",
                  "type": "p",
                },
              ],
              "type": "section",
            },
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Enter your message here
</h4>,
                <textarea
                  className="chatbox"
                  cols="100"
                  rows="10"
/>,
                <div
                  className="btn-group"
>
                  <button
                                    onClick={[Function]}
                  >
                                    Chat! 
                  </button>
</div>,
              ],
              "className": "chatinput",
              "name": "chatinput",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Enter your message here",
                },
                "ref": null,
                "rendered": "Enter your message here",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "chatbox",
                  "cols": "100",
                  "rows": "10",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    onClick={[Function]}
>
                    Chat! 
</button>,
                  "className": "btn-group",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Chat! ",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Chat! ",
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Chat Component should render initial layout 1`] = `
Array [
  <div>
    <div
        className="style"
    >
        <section
            className="chatmessages"
        >
            <section
                className="chatboxright"
            >
                <p
                    className="name"
                >
                    chef_name
                </p>
                <p
                    className="message"
                >
                    I am your chef today. What can i do for you?
                </p>
            </section>
        </section>
        <div
            className="chatinput"
            name="chatinput"
        >
            <h4>
                Enter your message here
            </h4>
            <textarea
                className="chatbox"
                cols="100"
                rows="10"
            />
            <div
                className="btn-group"
            >
                <button
                    onClick={[Function]}
                >
                    Chat! 
                </button>
            </div>
        </div>
    </div>
</div>,
]
`;
